compaction/compaction.cc

```diff
@@ -477,6 +477,7 @@ class compaction {
     std::unordered_set<shared_sstable> _compacting_for_max_purgeable_func;
     // optional owned_ranges vector for cleanup;
     owned_ranges_ptr _owned_ranges = {};
+    mutable std::optional<dht::incremental_owned_ranges_checker> _owned_ranges_checker;
```

Comment:
This mutable is suspicious. I'm not saying its wrong, but I want to understand why a checker is mutable. It seems like a very const thing.

Reply:
It's set by `make_sstable_reader() const`.  The patch didn't change that, just moved the member from `cleanup_compaction` to the base class.

Alternatively, `make_sstable_reader` can cheat and typecast `this` to a mutable pointer to set `_owned_ranges_checker`.

Reply:
done in v2

Reply:
"make_sstable_reader can cheat and typecast this to a mutable pointer to set _owned_ranges_checker." mutable keyword was introduced to simplify this :-)

Reply:
I submitted https://github.com/scylladb/scylladb/pull/12634 to mark belongs_to_current_node as const so we can remove the mutable definition in cleanup_compaction